src/main/java/seedu/duke/parser/journal/ParserJournal.java

```diff
@@ -81,195 +90,273 @@ public static boolean isValidNotebookCommand(String input) throws EmptyNoteArgum
-            String[] noteEntryNames = parseArgumentsDeleteEntryCommand(input);
-            int flagNotebook = notes.stream().anyMatch(note -> note.getNoteName().equals(noteEntryNames[0])) ? 1 : 0;
-            if (flagNotebook == 0) {
-                throw new NotebookNotFoundForEntry();
-            } else {
-                return new String[]{noteEntryNames[0], noteEntryNames[1]};
-            }
-        } else if (isEntryArgumentPresent && !isNoteArgumentPresent) {
-            throw new NotebookArgumentNotFoundException("Note argument not found!");
-        } else if (!isEntryArgumentPresent && isNoteArgumentPresent) {
-            throw new NotebookArgumentNotFoundException("Entry argument not found!");
+        String[] noteEntryNames = parseArgumentsDeleteEntryCommand(input);
+        int flagNotebook = notes.stream().anyMatch(note -> note.getNoteName().equals(noteEntryNames[0])) ? 1 : 0;
+        if (flagNotebook == 0) {
+            throw new NotebookNotFoundForEntry();
+        } else {
+            return new String[]{noteEntryNames[0], noteEntryNames[1]};
         }
-        return null;
     }
 
     /**
      * Parsing notebook and entry name.
      *
      * @param input from user
      * @return notebook name and entry name in form of string array
+     * @throws EmptyEntryArgumentsException if no arguments found after entry
+     * @throws EmptyNoteNameException if no note name found after n/
+     * @throws EmptyEntryNameException if no entry name found after e/
+     * @throws InvalidAddEntryArgumentException if arguments for adding entry are invalid.
      */
-    public static String[] parseNoteEntryName(String input) {
-        String noteNameDetails = input.trim().split("entry")[1];
-        String noteAndEntryName = noteNameDetails.split("n/")[1].trim();
-        String entryName = noteAndEntryName.split("e/")[1].trim();
-        String noteName = noteAndEntryName.split("e/")[0].trim();
-        return new String[]{noteName, entryName};
+    public static String[] parseNoteEntryName(String input) throws EmptyNoteNameException, EmptyEntryNameException,
+            EmptyEntryArgumentsException, InvalidAddEntryArgumentException {
+        String noteNameAndEntryNameDetails = input.trim().substring(input.indexOf("entry"));
+        if (input.trim().substring(input.indexOf("entry") + 5).trim().isEmpty()) {
+            throw new EmptyEntryArgumentsException();
+        }
+        String noteName = "";
+        String entryName = "";
+        String noteNameAndEntryName = "";
+        try {
+            noteNameAndEntryName =
+                    noteNameAndEntryNameDetails.substring(noteNameAndEntryNameDetails.indexOf("n/")).trim();
+        } catch (Exception e) {
+            throw new InvalidAddEntryArgumentException();
+        }
+
+        try {
+            noteName = noteNameAndEntryName.substring(noteNameAndEntryName.indexOf("n/") + 2,
+                    noteNameAndEntryName.indexOf("e/")).trim();
```

Comment:
The statement seem a bit lengthy. Consider extracting out index of e/ and n/ since you use these repeatedly.

Reply:
Yes. Will rectify in next PR.